server/src/Infrastructure/Persistence/Repositories/CourseRepository.cs

```diff
@@ -17,4 +18,13 @@ public async Task<List<Lecture>> GetLecturesByCourse(int courseId)
 
         return course.Lectures;
     }
+    public async Task DeleteByIdChecked(int courseId, string applicationUserId)
+    {
+        var course = await GetByIdChecked(courseId);
+        if (!course.CreatedBy.User.Equals(applicationUserId))
```

Comment:
maybe this logic should be in service? i think repository is only responsible for saving the entities, and this is business logic